spring-ai-core/src/main/java/org/springframework/ai/chat/client/ChatClient.java

```diff
@@ -216,9 +216,9 @@ interface ChatClientRequestSpec {
 
 		ChatClientRequestSpec tools(String... toolNames);
 
-		ChatClientRequestSpec tools(Object... toolObjects);
+		ChatClientRequestSpec tools(FunctionCallback... toolCallbacks);
```

Comment:
The Java compiler complaints when trying to pass an array of FunctionCallback and requires casting. In this way, we keep the same API, but it's more explicit and no casting is required.